### PR TITLE
109 test coverage

### PR DESCRIPTION
**Pull request by JPrevost**
This series of commits introduces Test Coverage reporting and begins the slog of work to actually write some meaningful tests.

This should bring us to about 24% test coverage. The models are about 60% tested.

It will start to get tricky doing tests of the workers and portions of the UI as so much relies on external HTTP calls. Capturing the results of HTTP calls once and replaying during testing is how I've handle this before and it looks like this project may be of use in that direction:
https://github.com/robfletcher/betamax

### CODE REVIEW

File: app/models/Item.scala [hunk 1]
```diff
@@ -319,7 +319,7 @@ object Item {
 
   def deleteBefore(date: Date) {
     DB.withConnection { implicit c =>
-      val rows = SQL("select id from item where created < {created}").on('created -> date)
+      val rows = SQL("select id from item where created <= {created}").on('created -> date)
```
Comment (JPrevost) [inline app/models/Item.scala]: This is one of the few functional changes in this series of commits. I believe it reflects the likely intent of the original code, but if not I can change it back and update the test to reflect that the date value being passed in will not actually be deleted.

Comment (richardrodgers) [inline app/models/Item.scala]: Seems fine - although in practice there is unlikely to be much difference - the granularity of Date in the DB is milliseconds - so very few ever be equal

**Comment by richardrodgers** [comment]
Although there is a lot to consider about how to test some of the code, this starts us down the road to a much higher confidence codebase (even finishing model unit tests). Thanks for making the effort!

:+1:

File: app/models/Item.scala (hunk 1 above)
Comment (JPrevost) [inline app/models/Item.scala]: Yeah. You are correct that it likely makes no real world difference.

**Pull request merged**